src/aposteriori/data_prep/create_frame_data_set.py

```diff
@@ -130,6 +134,71 @@ def encode_atom(self, atom_label: str) -> np.ndarray:
 
         return encoded_atom
 
+    def encode_gaussian_atom(
+        self, atom_label: str, modifiers_triple: t.Tuple[float, float, float]
+    ) -> (np.ndarray, int):
+        """
+        Encodes atom as a 3x3 gaussian with length of encoder length. Only the
+        C, N and O atoms are represented in gaussian form. If Ca and Cb are
+        encoded separately, they will be represented as a gaussian in their
+        separate channel.
+
+        Parameters
+        ----------
+        atom_label: str
+            Label of the atom to be encoded.
+        modifiers_triple: t.Tuple[float, float, float]
+            Triple of the difference between the discretized coordinate and the
+            undiscretized coordinate.
+
+        Returns
+        -------
+        atom_encoding: np.ndarray
+            Boolean array with atom encoding of shape (3, 3, 3, encoder_length,)
+
+        """
+        # Creating empty atom encoding:
+        encoded_atom = np.zeros((3, 3, 3, self.encoder_length))
+        # Attempt encoding:
+        if atom_label.upper() in self.label_to_encoding.keys():
+            atom_idx = self.label_to_encoding[atom_label.upper()]
+        # Fix labels:
+        elif self.encoder_length == 3:
+            # In this scenario, the encoder is C,N,O so Cb and Ca are just C atoms
+            if (atom_label.upper() == "CA") or (atom_label.upper() == "CB"):
+                atom_label = "C"
+                atom_idx = self.label_to_encoding[atom_label]
+        elif self.encoder_length == 4:
+            # In this scenario, Ca is a carbon atom but Cb has a separate channel
+            if atom_label.upper() == "CA":
+                atom_label = "C"
+                atom_idx = self.label_to_encoding[atom_label]
+        else:
+            raise ValueError(
+                f"{atom_label} not found in {self.atomic_labels} encoding. Returning empty array."
```

Comment:
I think the "Returning" part is left over from when it was an error. This won't return anything, as execution is halted by the error unless caught in a `try` statement.